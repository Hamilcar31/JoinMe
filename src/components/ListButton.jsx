import React from 'react';
import { ImageBackground, TouchableOpacity, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';

const list = require('../assets/list.png');

export default function ListButton(props) {
  const {
    onPress = () => {
    }, style,
  } = props;

  return (
    <TouchableOpacity onPress={() => onPress()} style={style}>
      <ImageBackground
        source={list}
        style={{
          zIndex: 3,
          width: 30,
          height: 30,
          shadowColor: 'black',
          shadowOffset: { height: 10 },
          shadowOpacity: 1,
          overflow: 'visible',
        }}
      />
    </TouchableOpacity>
  );
}

ListButton.propTypes = {
  onPress: PropTypes.func,
  style: ViewPropTypes.style,
};

ListButton.defaultProps = {
  onPress: () => {},
  style: {},
};
