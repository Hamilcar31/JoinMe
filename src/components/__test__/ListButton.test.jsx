import React from 'react';
import renderer from 'react-test-renderer';

import ListButton from '../ListButton';

describe('List button component', () => {
  it('with default parameters', () => {
    // given
    const listButton = <ListButton />;

    // when
    const tree = renderer.create(listButton).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
