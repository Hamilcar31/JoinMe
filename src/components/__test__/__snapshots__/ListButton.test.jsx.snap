// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List button component with default parameters 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 30,
        "overflow": "visible",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 10,
        },
        "shadowOpacity": 1,
        "width": 30,
        "zIndex": 3,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 30,
            "width": 30,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
